ql/src/test/org/apache/hadoop/hive/ql/udf/generic/TestGenericUDFFromUnixTime.java

```diff
@@ -0,0 +1,132 @@
+ */
+
+package org.apache.hadoop.hive.ql.udf.generic;
+
+import java.time.ZoneId;
+import org.apache.hadoop.hive.common.type.Date;
+import org.apache.hadoop.hive.common.type.Timestamp;
+import org.apache.hadoop.hive.common.type.TimestampTZ;
+import org.apache.hadoop.hive.common.type.TimestampTZUtil;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.ql.exec.MapredContext;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDF.DeferredJavaObject;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDF.DeferredObject;
+import org.apache.hadoop.hive.serde2.io.DateWritableV2;
+import org.apache.hadoop.hive.serde2.io.TimestampWritableV2;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.Text;
+
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+import org.apache.hadoop.mapred.JobConf;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.DateTimeFormatterBuilder;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+/**
+ * TestGenericUDFFromUnixTime.
+ */
+public class TestGenericUDFFromUnixTime {
+
+  public static void runAndVerify(GenericUDFFromUnixTime udf,
+      Object arg, Object expected) throws HiveException {
+    DeferredObject[] args = { new DeferredJavaObject(arg) };
+    Object result = udf.evaluate(args);
+    if (expected == null) {
+      assertNull(result);
+    } else {
+      assertEquals(expected.toString(), result.toString());
+    }
+  }
+
+  public static void runAndVerify(GenericUDFFromUnixTime udf,
+      Object arg1, Object arg2, Object expected) throws HiveException {
+    DeferredObject[] args = { new DeferredJavaObject(arg1), new DeferredJavaObject(arg2) };
+    Object result = udf.evaluate(args);
+
+    if (expected == null) {
+      assertNull(result);
+    } else {
+      assertEquals(expected.toString(), result.toString());
+    }
+  }
+
+  @Test
+  public void testTimestampDefaultTimezone() throws HiveException {
+    ObjectInspector valueLongOI = PrimitiveObjectInspectorFactory.writableLongObjectInspector;
+    GenericUDFFromUnixTime udf = new GenericUDFFromUnixTime();
+    ObjectInspector args[] = {valueLongOI};
+    udf.initialize(args);
+
+    Timestamp ts = Timestamp.valueOf("1470-01-01 00:00:00");
+    TimestampTZ tstz = TimestampTZUtil.convert(ts, ZoneId.systemDefault());
+
+    runAndVerify(udf,
+            new LongWritable(tstz.getEpochSecond()), new Text("1470-01-01 00:00:00"));
+
+    // test null values
+    runAndVerify(udf, null, null);
+  }
+
+  @Test
+  public void testTimestampOtherTimezone() throws HiveException {
+    ObjectInspector valueLongOI = PrimitiveObjectInspectorFactory.writableLongObjectInspector;
```

Comment:
Can we have same test with PrimitiveObjectInspectorFactory.writableIntObjectInspector as well?

Reply:
IntObjectInspector is removed from the code. so no need to add anything for that in the test.